src/Symfony/Component/Security/Http/Tests/Session/SessionRegistryTest.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Tests\Session;
+
+use Symfony\Component\Security\Http\Session\SessionRegistry;
+
+/**
+ * @author Antonio J. García Lagar <aj@garcialagar.es>
+ */
+class SessionRegistryTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGetAllSessionsInformation()
+    {
+        $storage = $this->getSessionRegistryStorage();
+        $storage->expects($this->once())->method('getAllSessionsInformation')->with('foo', true);
+        $registry = $this->getSessionRegistry($storage);
+        $registry->getAllSessionsInformation('foo', true);
+    }
+
+    public function testGetSessionInformation()
+    {
+        $storage = $this->getSessionRegistryStorage();
+        $storage->expects($this->once())->method('getSessionInformation')->with('foobar');
+        $registry = $this->getSessionRegistry($storage);
+        $registry->getSessionInformation('foobar');
+    }
+
+    public function testUpdateLastUsed()
+    {
+        $sessionInformation = $this->getSessionInformation();
+        $sessionInformation->expects($this->once())->method('updateLastUsed');
+        $storage = $this->getSessionRegistryStorage();
+        $storage->expects($this->any())->method('getSessionInformation')->with('foobar')->will($this->returnValue($sessionInformation));
+        $storage->expects($this->once())->method('setSessionInformation')->with($sessionInformation);
+        $registry = $this->getSessionRegistry($storage);
+        $registry->updateLastUsed('foobar', time() - 5);
+    }
+
+    public function testExpireAt()
+    {
+        $sessionInformation = $this->getSessionInformation();
+        $sessionInformation->expects($this->once())->method('expireAt');
+        $storage = $this->getSessionRegistryStorage();
+        $storage->expects($this->any())->method('getSessionInformation')->with('foobar')->will($this->returnValue($sessionInformation));
+        $storage->expects($this->once())->method('setSessionInformation')->with($this->identicalTo($sessionInformation));
+        $registry = $this->getSessionRegistry($storage);
+        $registry->expireAt('foobar', time() - 5);
```

Comment:
Hi. Firsts forgive for reviving a very old PR. @timglabisch I indeed have a use case scenario where a concurrent session control (invalidation in my case) would be needed. I'm developing an application that's based in the ussual mulpiple license types schema. One license is for just one user, another one is for 5 users, etc. I don't want to let a user be able to login in different places at the same.